src/app/containers/ArticleTimestamp/shouldDisplayLastUpdatedTimestamp.test.jsx

```diff
@@ -0,0 +1,130 @@
+      const currentTime = originalDate();
+      const twoMinutesAgo = timestampGenerator({ minutes: 2 });
+      const msDifference = currentTime - twoMinutesAgo;
+      const timeDifferenceMinutes = msDifference / 1000 / 60;
+
+      expect(
+        hasBeenUpdated({ timeDifferenceMinutes, minutesTolerance }),
+      ).toEqual(true);
+    });
+
+    it('should return false when the time difference between firstPublished and lastPublished in minutes is less than the minutes tolerance', () => {
+      const currentTime = originalDate();
+      const fortyEightSecondsAgo = timestampGenerator({ minutes: 0.8 });
+      const msDifference = currentTime - fortyEightSecondsAgo;
+      const timeDifferenceMinutes = msDifference / 1000 / 60;
+
+      expect(
+        hasBeenUpdated({ timeDifferenceMinutes, minutesTolerance }),
+      ).toEqual(false);
+    });
+  });
+
+  describe('publishedAndUpdatedToday', () => {
+    it('should return true when firstPublished and lastPublished today', () => {
+      const [midnightToday, oneAmToday] = sameDayTimestampsGenerator({
+        intervals: [{ hours: 1 }],
+      });
+      const wasPublishedAndUpdatedToday = publishedAndUpdatedToday({
+        firstPublished: midnightToday,
+        lastPublished: oneAmToday,
+      });
+
+      expect(wasPublishedAndUpdatedToday).toEqual(true);
+    });
+
+    it('should return false when firstPublished is not today', () => {
+      const oneDayAndThreeMinutesAgo = timestampGenerator({
+        days: 1,
+        minutes: 3,
+      });
+      const oneDayAndOneMinuteAgo = timestampGenerator({ days: 1, minutes: 1 });
+      const wasPublishedAndUpdatedToday = publishedAndUpdatedToday({
+        firstPublished: oneDayAndThreeMinutesAgo,
+        lastPublished: oneDayAndOneMinuteAgo,
+      });
+
+      expect(wasPublishedAndUpdatedToday).toEqual(false);
+    });
+  });
+
+  describe('shouldDisplayLastUpdatedTiemstamp', () => {
+    it('should return true when article was published and updated today', () => {
+      const [midnightToday, oneAmToday] = sameDayTimestampsGenerator({
+        intervals: [{ hours: 1 }],
+      });
+      const shouldLastUpdatedTimestampBeDisplayed = shouldDisplayLastUpdatedTimestamp(
+        {
+          minutesTolerance,
+          firstPublished: midnightToday,
+          lastPublished: oneAmToday,
+        },
+      );
+
+      expect(shouldLastUpdatedTimestampBeDisplayed).toEqual(true);
+    });
+
+    it('should return true when article was lastUpdated within relative time period', () => {
+      const nineHoursAgo = timestampGenerator({ hours: 9 });
+      const currentTime = originalDate();
+      const shouldLastUpdatedTimestampBeDisplayed = shouldDisplayLastUpdatedTimestamp(
+        {
+          minutesTolerance,
+          firstPublished: nineHoursAgo,
+          lastPublished: currentTime,
+        },
+      );
```

Comment:
Good point - because `publishedAndUpdatedToday` will evaluate to true after 09:00 and therefore so will `shouldLastUpdatedTimestampBeDisplayed` regardless. Hmmn. 

(I think) using the relative/current time is significant here - as the `isLastRelative` function uses Date.now() as its point of comparison for finding out if a time period is <10hrs ago. Even if we fabricate a firstPublished time that is on a previous day, nine hours before a lastPublished time, the `isLastRelative` function will be comparing lastPublished against Date.now().

That logic ('isLastRelative') has been separately tested within `helpers.test.js`. **This** test (`'should return true when article was lastUpdated within relative time period'`) was to test its implementation within the composite function - but I'm not sure we can.

My gut says to just remove it - thoughts?

Reply:
Makes sense - happy to go with your gut!

Reply:
Actually think I found a better solution - mocking Date.now within the jest file to be 08:30am. 

All other tests work as expected and with a couple of modifications - the test in question is now checking that the second timestamp should be rendered for an article firstPublished and lastUpdated yesterday (i.e. the same day && not today - therefore would not render without the condition being tested), while the current time is within the lastPublished relative time period (i.e. updatedWithinRelativeTimePeriod  === true, the thing we want this test to be checking).